### PR TITLE
Remove char_u (3): Replace char_u in vim_version_dir

### PR DESCRIPTION
**Pull request by mbainter**
See: #459

### CODE REVIEW

File: src/nvim/os/env.c [hunk 1]
```diff
@@ -371,19 +371,19 @@ void expand_env_esc(char_u *srcp, char_u *dst, int dstlen, bool esc, bool one,
...
 /// @param vimdir directory to test
-static char_u *vim_version_dir(char_u *vimdir)
+static char *vim_version_dir(char *vimdir)
 {
   char_u      *p;
 
   if (vimdir == NULL || *vimdir == NUL)
     return NULL;
-  p = concat_fnames(vimdir, (char_u *)VIM_VERSION_NODOT, true);
+  p = concat_fnames((char_u *) vimdir, (char_u *)VIM_VERSION_NODOT, true);
```
Comment (elmart) [inline src/nvim/os/env.c]: Unneeded extra space after cast.

Comment (mbainter) [inline src/nvim/os/env.c]: Fixed

**Comment by elmart** [comment]
LGTM.

**Comment by justinmk** [comment]
Merged, thanks.

@mbainter I suggest using `remove char_u: ...` as a context prefix in the commit messages for these various char_u refacorings. There will be many of them... thanks

**Comment by elmart** [comment]
In fact, I would suggest the following structure:
PR title: `Remove char_u (<number>).`
Commit title: `Remove char_u: function_being_refactored().`

**Comment by justinmk** [comment]
:+1:

**Comment by Pyrohh** [comment]
@mbainter do you plan on touching anything besides what you mentioned in https://github.com/neovim/neovim/pull/2413#issuecomment-92100164 in your next PR? I'd like to work on this, and don't want to clash with you.

**Comment by Pyrohh** [comment]
@mbainter nevermind, I have too many doc related cleanups I'm working on.

**Comment by mbainter** [comment]
I'm just going to keep hacking away at it.  You can chime in anytime, feel free to ping me and I'll tell you what I have active at the time.  Right now everything is around the (get|set)env() functions, and what they call.  Next in my queue at the moment is modname(), make_percent_swname(), and makeswapname().

**Comment by Pyrohh** [comment]
Alright, sounds good. I'm glad you're doing them in small batches; whenever I attempt to do these conversions the next thing I know is I have a diff spanning 10 files and 500 lines :grin:

**Pull request closed without merging**